slack.cpp

```diff
@@ -63,35 +63,74 @@ SimpleHttps	*https;
 	pair<string, string> header = make_pair("Content-type", "application/json");
 	headers.push_back(header);
 
+	if (m_url.empty())
```

Comment:
Main issue has been resolved now but I think, here you should also trim it first. If I put `space` in the field, then in syslog-

`Sep  3 16:44:12 foglamp-ThinkPad-E450 FogLAMP Notifications[16987]: ERROR: Exception while sending notification to slack webhook   : basic_string::substr: __pos (which is 2) > this->size() (which is 1)`

Reply:
Let's create another JIRA task for this and close this